src/main/java/dev/morling/onebrc/CalculateAverage_royvanrijn.java

```diff
@@ -15,65 +15,309 @@
  */
 package dev.morling.onebrc;
 
+import java.io.File;
 import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.channels.FileChannel;
 import java.nio.file.Files;
 import java.nio.file.Path;
-import java.util.AbstractMap;
-import java.util.Map;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.TreeMap;
 import java.util.stream.Collectors;
 
+/**
+ * Changelog:
+ *
+ * Initial submission:          62000 ms
+ * Chunked reader:              16000 ms
+ * Optimized parser:            13000 ms
+ * Branchless methods:          11000 ms
+ * Adding memory mapped files:  6500 ms (based on bjhara's submission)
+ * Skipping string creation:    4700 ms
+ * Custom hashmap...            4200 ms
+ * Added SWAR token checks:     3900 ms
+ * Skipped String creation:     3500 ms (idea from kgonia)
+ * Improved String skip:        3250 ms
+ * Segmenting files:            3150 ms (based on spullara's code)
```

Comment:
I believe EOL is small enough distant that it is not worthy, unless you first check for it.
A trick to make EOL faster is to explode the loop search , as I have done in Netty (that was a the unrolled version for the tail checks < 8 bytes) - that was the reason for that, and was the reason why the unrolled version was coming "before" the batchy one ie to speedup < 8 bytes length cases

Reply:
I see you already have an enrolled check there, so I think my previous comment is unnecessary